src/Interceptor/RetryRequests.php

```diff
@@ -27,23 +24,21 @@ public function __construct(int $retryLimit)
     public function request(Request $request, CancellationToken $cancellation, Client $client): Promise
     {
         return call(function () use ($request, $cancellation, $client) {
-            $attempts = 1;
+            $attempt = 1;
 
             do {
                 try {
                     return yield $client->request(clone $request, $cancellation);
                 } catch (UnprocessedRequestException $exception) {
                     // Request was deemed retryable by connection, so carry on.
-                } catch (InvalidRequestException | ParseException | TimeoutException $exception) {
-                    // Request is or response is invalid or request timed out, so do not retry.
-                    throw $exception;
-                } catch (HttpException $exception) {
+                } catch (SocketException $exception) {
```

Comment:
For this to work, I think `Http2Exception` needs to extend `SocketException`.

Reply:
I've looked through the list of exceptions and most of them seem to be protocol violations, no?

Reply:
Yes. I suppose if the server violates the protocol, retrying may not help.